### PR TITLE
chore(text-field): moved outlined text-field sass rules into private mixins

### PR DESCRIPTION
**Pull request by moog16**
partial fix: #1598

### CODE REVIEW

File: packages/mdc-textfield/_mixins.scss [hunk 1]
```diff
@@ -180,6 +180,132 @@
...
+  height: 48px;
+
+  .mdc-text-field__input {
+    padding: 12px 12px 7px;
+  }
+
+  .mdc-text-field__label {
+    bottom: 18px;
+
+    &--float-above {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: make this `.mdc-text-field__label--float above` ... drop the `&`

File: packages/mdc-textfield/_mixins.scss [hunk 2]
```diff
@@ -180,6 +180,132 @@
...
+      transform: translateY(calc(-122% - 2px)) scale(.923);
+
+      // stylelint-disable max-nesting-depth
+      // stylelint-disable-next-line selector-max-specificity
+      &.mdc-text-field__label--shake {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: drop the `&` syntax here as well...too confusing

File: packages/mdc-textfield/_mixins.scss [hunk 3]
```diff
@@ -180,6 +180,132 @@
...
+  }
+
+  .mdc-text-field__label {
+    @include mdc-rtl-reflexive-position(left, 16px);
+
+    position: absolute;
+    bottom: 20px;
+    transition: transform 260ms ease;
+
+    &--float-above {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: drop the & syntax here as well...too confusing

File: packages/mdc-textfield/_mixins.scss [hunk 4]
```diff
@@ -180,6 +180,132 @@
...
+      transform: scale(.75) translateY(-170%);
+
+      &.mdc-text-field__label--shake {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: drop the & syntax here as well...too confusing

File: packages/mdc-textfield/_mixins.scss [hunk 5]
```diff
@@ -180,6 +180,132 @@
...
+
+  .mdc-text-field__icon {
+    z-index: 2;
+
+    &:hover ~ .mdc-text-field__idle-outline {
+      border: 1px solid $mdc-text-field-outlined-hover-border;
+    }
+  }
+
+  &.mdc-text-field--focused {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: Don't put this within the`mdc-text-field-outlined` mixin, just call it directly from `mdc-text-field.scss`

File: packages/mdc-textfield/_mixins.scss [hunk 6]
```diff
@@ -180,6 +180,132 @@
...
+
+  .mdc-text-field__label--float-above ~ .mdc-text-field__idle-outline {
+    opacity: 0;
+  }
+
+  .mdc-text-field__label--float-above ~ .mdc-text-field__outline {
+    opacity: 1;
+  }
+
+  &.mdc-text-field--disabled {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: Don't put this within the`mdc-text-field-outlined` mixin, just call it directly from `mdc-text-field.scss`

File: packages/mdc-textfield/_mixins.scss [hunk 7]
```diff
@@ -180,6 +180,132 @@
...
+    @include mdc-text-field-outlined-disabled_;
+  }
+
+  &.mdc-text-field--dense {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/_mixins.scss]: Don't put this within the`mdc-text-field-outlined` mixin, just call it directly from `mdc-text-field.scss`

File: packages/mdc-textfield/mdc-text-field.scss [hunk 8]
```diff
@@ -177,37 +189,39 @@
...
+  }
 
-    &--float-above {
-      transform: translateY(-50%) scale(.75, .75);
-    }
+  .mdc-text-field__label--float-above {
+    transform: translateY(-50%) scale(.75, .75);
   }
 
   .mdc-text-field__label--float-above {
```
Comment (lynnjepsen) [inline packages/mdc-textfield/mdc-text-field.scss]: nit: you can combine this with the `.mdc-text-field__label--float-above` above

Comment (moog16) [reply packages/mdc-textfield/mdc-text-field.scss]: will fix in another PR after this

**Pull request merged**